lib/transport/binary/protocol28/operation.js

```diff
@@ -394,10 +396,49 @@ Operation.prototype.consume = function (buffer, offset) {
 
     code = buffer.readUInt8(offset);
     if (code === 3) {
-      offset += 5; // ignore the next integer
-      obj = {};
-      offset += this.parsePushedData(buffer, offset, obj, 'data');
-      return [Operation.PUSH_DATA, offset, obj.data];
+      var sessionId = buffer.readInt32BE(offset+1);
+      var pushType = buffer.readUInt8(offset + 5);
+
+      if (pushType === 81) {
+        offset += 6; // ignore the next integer
+        var length = buffer.readInt32BE(offset);
+        offset+=4;
+        var operation = buffer.readUInt8(offset);
+        offset+=1;
+        var token = buffer.readInt32BE(offset);
+        offset+=4;
+        var recordType = buffer.readUInt8(offset);
+        offset+=1;
+        var version = buffer.readInt32BE(offset);
+        offset+=4;
+        var clusterId = buffer.readInt16BE(offset);
+        offset+=2;
+        var clusterPosition = Long.fromBits(
+            buffer.readUInt32BE(offset + 4),
+            buffer.readInt32BE(offset)
+        ).toNumber();
+        offset+=8;
+        var contentLenght = buffer.readInt32BE(offset);
+        offset+=4;
+
+        var asString = buffer.toString('utf8', offset, offset + contentLenght);
+        offset += contentLenght;
+        var content = deserializer.deserialize(asString, this.data.transformerFunctions);
+
+        var obj = {};
```

Comment:
Hi Charles,

thank you very much, I'll fix it and will send another PR soon

Thanks

Luigi


2015-04-27 20:42 GMT+02:00 Charles Pick <notifications@github.com>:

> In lib/transport/binary/protocol28/operation.js
> <https://github.com/codemix/oriento/pull/306#discussion_r29177038>:
>
> > +        offset+=4;
> > +        var clusterId = buffer.readInt16BE(offset);
> > +        offset+=2;
> > +        var clusterPosition = Long.fromBits(
> > +            buffer.readUInt32BE(offset + 4),
> > +            buffer.readInt32BE(offset)
> > +        ).toNumber();
> > +        offset+=8;
> > +        var contentLenght = buffer.readInt32BE(offset);
> > +        offset+=4;
> > +
> > +        var asString = buffer.toString('utf8', offset, offset + contentLenght);
> > +        offset += contentLenght;
> > +        var content = deserializer.deserialize(asString, this.data.transformerFunctions);
> > +
> > +        var obj = {};
>
> obj is defined twice in this function so the tests are failing, I would
> remove the obj declaration and replace this with:
>
> return [
>   Operation.LIVE_RESULT,
>   token,
>   operation,
>   {
>     content: content,
>     type: 'd',
>     cluster: clusterId,
>     position: clusterPosition,
>     version: version
>   },
>   offset
> ];
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codemix/oriento/pull/306/files#r29177038>.
>